### PR TITLE
Collapse student registration pages into one page

### PR DESCRIPTION
**Pull request by wishdasher**
* Moved registration and other program-specific routes from `App` to `RegDashboard` (which functions as a landing page for a specific program)
* Lifted registration progress state out of `StudentRegDashboard` into `RegDashboard`
* Moved registration steps into a rudimentary implementation of registration flow in `StudentRegistration`

Long PR, sorry, many of the changes were interdependent and would not have worked without duplicating a lot of code (from `StudentRegDashboard` to `StudentRegistration`)

Many TODOs about things we need to improve with this system

### CODE REVIEW

**Comment by wishdasher** [comment]
image
image
image

File: client/src/registration/StudentRegistration.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          <span className="steps-marker"></span>
+          <div className="steps-content">
+            <p className="is-size-4">{stepText}</p>
+          </div>
+        </a>
+      </li>
+    );
+  }
+
+  console.log(props.checks);
```
Comment (evantey14) [inline client/src/registration/StudentRegistration.tsx:83]: nit: remove?

File: client/src/registration/RegDashboard.tsx [hunk 2]
```diff
@@ -11,14 +17,59 @@ type Props = {
   edition: string;
 };
 
+// TODO: restructure this per-program "dashboard" concept, if anything, this
+// component is functioning more like a "ProgramDashboard"
```
[the remark this answers is not in the document]
Comment (mvadari) [reply client/src/registration/RegDashboard.tsx:21]: ++

It's not even just student registration - it's all of registration.

File: client/src/registration/RegDashboard.tsx [hunk 3]
```diff
@@ -11,14 +17,59 @@ type Props = {
...
+
   if (loggedIn) {
     return (
+      // TODO: create program-specific context and provider to subsequent components
       <div className="container">
-        {isStudent && <StudentRegDashboard program={props.program} edition={props.edition} />}
+        {isStudent && (
+          // TODO: overarching application routing organization in one place
+          <Router>
+            <Redirect from="/" to="dashboard" noThrow />
```
[the remark this answers is not in the document]
Comment (wishdasher) [reply client/src/registration/RegDashboard.tsx:57]: So dashboard currently shows up on e.g. /splash/2019, as a placeholder.

File: client/src/registration/RegDashboard.tsx [hunk 4]
```diff
@@ -11,14 +17,59 @@ type Props = {
...
+            <StudentRegDashboard
+              /* @ts-ignore TODO: thanks reach router */
```
[the remark this answers is not in the document]
Comment (wishdasher) [reply client/src/registration/RegDashboard.tsx:59]: Same issue as in `App.tsx`. It's already a trello task.

File: client/src/registration/RegDashboard.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (wishdasher) [reply client/src/registration/RegDashboard.tsx:21]: > It's not even just student registration - it's all of registration.

That's a separate issue. I can forge ahead with the concept of registration I thought we'd discussed, but maybe there should be another one (discussion).

**Pull request merged**